gradle/javadoc.gradle

```diff
@@ -18,17 +18,18 @@ import static org.gradle.util.CollectionUtils.single
 
 
 def javaMainProjects = subprojects.findAll { !['util'].contains(it.name) }
-task docs(type: Javadoc) {
-    destinationDir = new File(rootDir, 'build/docs')
-    javaMainProjects.each { proj ->
-        proj.tasks.withType(Javadoc).each { javadocTask ->
-            source += javadocTask.source
-            classpath += javadocTask.classpath
-            excludes += javadocTask.excludes
-            includes += javadocTask.includes
-            dependsOn +=javadocTask.dependsOn
-        }
-    }
+
+task docs {
+    dependsOn subprojects.collect { it.tasks.withType(Javadoc) }
+    dependsOn subprojects.collect { it.tasks.withType(ScalaDoc) }
+}
+
+def buildDocsUrl(String version, String packageName) {
+    def docVersion = version.split('-').head()
+    def versionList = docVersion.split('\\.')
+    def patch = (versionList[1] as int) - 1
+    docVersion = patch >= 0 ? "${versionList[0]}.${patch}" : '4.0'
+    ["http://mongodb.github.io/mongo-java-driver/${docVersion}/apidocs/${packageName}/", "${rootProject.buildDir.path}/docs/${packageName}"]
```

Comment:
Can this be a relative URIs so that the links will work if they are published to a different root URL, e.g. `docs.mongodb.com`?

Reply:
Then all the javadoc.jars won't work. So it needs to be a real url.  I previously used javadoc.io but then changed it to be the github url.

Reply:
OK, so when this is published to javadoc.io, the links from, say, mongodb-driver-sync to bson, will go to mongodb.github.io.

It just means that when we migrate our docs off of gh-pages we'll have to update this.  I guess that's ok.